//
// Copyright 2018 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

enum HashAlgorithm {
  UNKNOWN_HASH_ALGORITHM = 0;

  // SHA-2 family of algorithms
  SHA224 = 1;
  SHA256 = 2;
  SHA384 = 3;
  SHA512 = 4;

  SHA_1 = 5;
}

enum AeadScheme {
  UNKNOWN_AEAD_SCHEME = 0;

  AES128_GCM = 1;
  AES256_GCM = 2;

  AES128_GCM_SIV = 3;
  AES256_GCM_SIV = 4;
}

enum SignatureScheme {
  UNKNOWN_SIGNATURE_SCHEME = 0;

  ECDSA_P256_SHA256 = 1;
  ECDSA_P384_SHA384 = 2;
}

enum AsymmetricEncryptionScheme {
  UNKNOWN_ASYMMETRIC_ENCRYPTION_SCHEME = 0;

  RSA2048_OAEP = 1;
  RSA3072_OAEP = 2;
}

enum AsymmetricKeyEncoding {
  UNKNOWN_ASYMMETRIC_KEY_ENCODING = 0;

  ASYMMETRIC_KEY_PEM = 1;
  ASYMMETRIC_KEY_DER = 2;
}
